File: back/src/main/java/com/sunghwa/board/service/CommentService.java
package com.sunghwa.board.service;

import com.sunghwa.board.domain.Comment;
import com.sunghwa.board.repository.CommentRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CommentService {
    private final CommentRepository commentRepository;

    public CommentService(CommentRepository commentRepository) {
        this.commentRepository = commentRepository;
    }

    public Comment addComment(Comment comment) {
        commentRepository.insert(comment);
        return comment;
    }

    public List<Comment> findOne(Long postId) {
        return commentRepository.findByPostId(postId);
    }

    public Comment updateComment(Comment comment) {
        commentRepository.update(comment);
        return comment;
    }

    public Long deleteComment(Long commentId) {
        commentRepository.delete(commentId);
        return commentId;
    }
}

File: front/src/api/CommentService.js
import api from '@/api/apiConfig';

const CommentService = {
  getCommentByPostId: async postId => {
    const response = await api.get(`/api/comment/${postId}`);
    return response.data;
  },

  addComment: async ({ writer, contents, postId }) => {
    const response = await api.post('/api/comment/add', {
      writer,
      contents,
      postId,
    });

    return response;
  },

  updateComment: async ({ contents, commentId }) => {
    const response = await api.post('/api/comment/config', {
      commentId,
      contents,
    });

    return response;
  },

  deleteComment: async commentId => {
    const response = await api.post('/api/comment/delete', {
      commentId,
    });

    return response;
  },
};

export default CommentService;
